packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.js

```diff
@@ -102,29 +102,35 @@ export const TagOverflowPopover = React.forwardRef(
           <PopoverContent>
             <div ref={overflowTagContent} className={`${blockClass}__content`}>
               <ul className={`${blockClass}__tag-list`}>
-                {getOverflowPopoverItems().map((tag) => {
-                  const tagProps = {};
-                  if (overflowType === 'tag') {
-                    tagProps.type = 'high-contrast';
+                {getOverflowPopoverItems().map(
+                  ({ tagType, onClose, ...other }) => {
+                    const tagProps = { type: tagType, ...other };
+                    if (overflowType === 'tag') {
+                      tagProps.type = 'high-contrast';
```

Comment:
same comment above regarding variable mutation. it would be preferable to just compute this value in the prop instead of overwriting anything.

_if_ we wanted to take this approach however i would say that it should be handled in the `getOverflowPopoverItems` function. so instead of putting this logic here, it could look more like
```
const getOverflowPopoverItems = () => {
  return overflowTags.filter( ... ).map( { code to add or change properties } );
};
```

Reply:
Since we already use a map method to render the array elements, I'd rather not introduce another map within getOverflowPopoverItems. So, like you mentioned in the previous comment, I used variables for the conditionals rather than mutating the object.